recipes/newrelic/apm/php/debian.yml

```diff
@@ -0,0 +1,502 @@
+          current_user=$SUDO_USER
+          for PROCESS in ${guided_support[@]}
+          do
+            #
+            # First look for the full process name with a minimal set of other values so
+            # that there is no chance of filtering off of those (for instance, when the
+            # username is `apache` or something similar to the filters.  Then use the name
+            # for other searches.
+            #
+            full_process_name=$(ps xf | egrep $PROCESS | grep -v grep | grep -v yml | head -n1 | awk '{print $5}')
+            priv_user=$(ps auxf | egrep $full_process_name | grep -v grep | grep -v yml | head -n1 | awk '{print $1}')
+            fpm_nginx_process_name=$(echo "${full_process_name}" | sed -n 's/\(.*\):$/\1/p')
+            if [ -n "${fpm_nginx_process_name}" ]; then
+              fpm_process_loc=$(ps auxf | egrep $full_process_name | grep -v grep | grep -v yml | head -n1 | awk '{print $14}')
+              fpm_ver=$(echo "${fpm_process_loc}" | sed -n 's/.*\/php\/\([578].[0-9]\)\/.*/\1/p')
+              if [ -n "${fpm_ver}" ]; then
+                # It's fpm.
+                process_name="php${fpm_ver}-fpm"
+              else
+                # It's nginx.
+                process_name=$fpm_nginx_process_name
+              fi
+            else
+              process_name=$(echo $full_process_name | sed -n 's/.*\/\(.*\)$/\1/p')
+            fi
+            nonpriv_user=
+            if [ -n "${priv_user}" ]; then
+              nonpriv_user=$(ps auxf | egrep $full_process_name | grep -v grep | grep -v yml | grep -v $priv_user | head -n1 | awk '{print $1}')
+            fi
+            if [ -n "${process_name}" ]; then
+              if [ -z "${nonpriv_user}" ]; then
+                nonpriv_user=$current_user
+              fi
+              #
+              # Output all processes to be restarted to a file with the following format:
+              # process_name privileged_user
+              # When the process is restarted, it will be used with the associated
+              # privileged user.
+              #
+              echo "$process_name $priv_user" >> {{.TMP_INSTALL_DIR}}/processes_to_restart.txt
+              #
+              # Save all non-privileged users associated with detected processes.
+              # When initiating the transaction, it will ONLY be used with
+              # the first associated non-privileged user because due the reasoning above
+              # that the installation logic adheres to, between FPM and Apache,
+              # FPM will be the one to use and `newrelic-install.sh` only
+              # installs the `newrelic.ini` in one web location.
+              #
+              echo "$nonpriv_user" >> {{.TMP_INSTALL_DIR}}/nonpriv_users.txt
+            fi
+          done
+
+    add_gnupg2_curl_if_required:
+      cmds:
+        - |
+          if [ $(({{.DEBIAN_VERSION}})) -ge 10 ]; then
```

Comment:
@zsistla 
We dont need to install curl, it's a valid assumption, users cant even download the CLI to get to the recipe install.
As for gnupg2, yes we should check if it's not already there before attempting to install all the time.
apt-get/yum install have a tendency to fail on older versions, because the user's host may have some outdated repos that will fail when doing an apt-get/yum update, which by the way I'm not seeing any update here.